### DISCUSSION TITLE
Is there a method or variable to access the time taken to solve a model?

### DISCUSSION THREAD

**Discussion by zbatool** [coin-or/Sonnet#14] [Q&A]
I am solving mps files using Sonnet with the OsiClpSolverInterface. I am storing some statistics in an external output csv such as solver.NumVariables(), model.name but haven't been able to find such a property or method for the time taken to solve the model. When solving the program does print out onto the terminal the time in the last line e.g.:
"Clp0032I Optimal objective xxx xx iterations time 0.002, Presolve 0.00"
but I want to able to access the time separately so I can feed it directly into an output file. Does any such method exist and I have just missed it?

**Comment by jhmgoossens** [comment]
Not as a property. Sonnet uses mostly plain Osi methods available, and time is not one of them. There is [Osi IterationCount](https://www.coin-or.org/Doxygen/Osi/classOsiSolverInterface.html#a195998db1a5da926130bd6fe93144ac3), available in Sonnet via solver.IterationCount, but that is probably just an indicator. Even the underlying Clp solver used by OsiClp does not (AFAIK) have properties for retrieving the time it took to solve. 
The Clp examples almost all use the [double CoinCpuTime()](https://github.com/coin-or/CoinUtils/blob/master/src/CoinTime.hpp) of CoinUtils to measure the time interval. See [Clp's addRows example](https://github.com/coin-or/Clp/blob/master/examples/addRows.cpp).
The CoinUtils CoinCpuTime() function is available in Sonnet via COIN.CoinUtils.CoinCpuTime() implemented in SonnetWrapper.
If you are looking for wall clock time, then just use .NET [System.Diagnostics.Stopwatch](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch?view=net-5.0)
Does this help?
